### ISSUE TITLE
[CDP] Improve error messaging for a better screen reader user experience

### ISSUE THREAD

**Issue by josephrlee** [department-of-veterans-affairs/va.gov-team#75547]
# [508-defect-4](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

Issue Title: 508 Defect 4, Screen Readers, Cognition, error message can be improved for better user experience.
Point of Contact: Angela Fowler
**VFS Point of Contact:** _First name only: Angela

## User Story or Problem Statement:
As a screen reader user, it is easier to navigate error messages when they contain an action item I.E. "return to my VA debt," and when focus is placed on said action item when the error is triggered. 

## Details
In the event that there are no debt letters associated with a user account an error message appears on the screen. The screen reader announces the message in full, and keyboard focus is placed on the h1. The user can return to the previous page by navigating up to the breadcrumb.
User experience can be improved by including a "Return" link within the error message and moving focus to it upon the triggering of the message. 

## Tasks
- [x] Come up with a list of pages that need links back to the portal 
- [x] Define which links should go where
- [x] Update CDP figma file to reflect new links
- [x] Create annotations for dev/handoff

## Acceptance Criteria
> Generated by [Zenhub AI](https://www.zenhub.com/ai)

- [x] **Scenario 1: Improve error messaging for screen reader users**
  - Given there is an error message displayed on the screen
  - When the error message is triggered
  - Then the error message should include a "Return" link
  - And the focus should be moved to the "Return" link

**Comment by josephrlee** [comment]
Let's add this to the next sprint! 

cc: @mgleo @charlottecesana

**Comment by HeatherRienks** [comment]
@Tom-Davis - FYI -do we want to add this to S124?

**Comment by Tom-Davis** [comment]
@HeatherRienks let's move it to ready and see what the landscape looks like at planning, but being in ready will keep it from getting forgotten.

**Comment by mgleo** [comment]
Messaging for the error on the Debt Letters pages needs to be improved.

<img width="1206" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNEYvQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--fa1ed7943b182064eb5444f8473b4430c16ef132/Screenshot%202024-04-22%20at%209.52.32%E2%80%AFAM.png" alt="Screenshot 2024-04-22 at 9.52.32 AM.png" />

**Comment by HeatherRienks** [comment]
Team is focused on VBA Payment History work and this ticket is looking like it will likely roll over into the next sprint as other work is taking priority.

**Comment by HeatherRienks** [comment]
Per Sprint Planning on 6/18, moving back to Ready while we focus on MVP - VBA Payment History.

**Comment by maxx1128** [comment]
I started looking at where the code needed to change this ticket would be. Many files within the `combined-debt-portal` directory in `vets-website` would potentially need to be changed: AlertCard.jsx, ComboAlerts.jsx, DisasterAlert.jsx, MCPAlerts.jsx, Alerts.jsx, DebtDetailsCard.jsx, DebtLettersSummary.jsx, and OverviewPage.jsx.

If the new error message needs a return link that automatically gets focused on, it'd be best to make another alert component based on the usual `va-alert` component. This component takes in the same arguments but has this extra functionality built in. That will make swapping the alert components where needed much easier.

**Comment by HeatherRienks** [comment]
This ticket was not pulled into the sprint, it was moved to "ready" (see comment from 6/18/24), but the sprint label was still attached to the ticket, so it appears that it rolled over.

**Comment by HeatherRienks** [comment]
In today's Sprint Planning, we committed to pulling this ticket into S129!

**Comment by Tom-Davis** [comment]
We are re scoping ticket [87478 ](https://app.zenhub.com/workspaces/vsa---debt-607736a6c8b7e2001084e3ab/issues/gh/department-of-veterans-affairs/va.gov-team/87478) and as a result this work is no longer needed. @maxx1128 & @josephrlee if we have anything in development that we need to depreciate let me know and we can address that with a smaller maintenance ticket.

**Comment by josephrlee** [comment]
@Tom-Davis, @denisecoveyduc, I've been thinking more about this ticket. It's not just for the debt letters but for the larger debt portal; there are quite a few other instances of alerts in the debt portal where there are "dead ends" that this solution could still be useful. I think we should re-open it and apply the solution I developed to the other pages and exclude the debt letters.

**Comment by josephrlee** [comment]
@maxx1128, I took a look at the files you mentioned, I think we only need to add the fix to update these 3 files:

- [ComboAlerts.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/combined-debt-portal/combined/components/ComboAlerts.jsx) that appears on combined debt portal [OverviewPage.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/combined-debt-portal/combined/containers/OverviewPage.jsx)
- Any [Alerts.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/combined-debt-portal/debt-letters/components/Alerts.jsx) that appears on [DebtLettersSummary.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/combined-debt-portal/debt-letters/containers/DebtLettersSummary.jsx)
- [alert-messages.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/combined-debt-portal/combined/utils/alert-messages.jsx) on the copay [OverviewPage.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/combined-debt-portal/medical-copays/containers/OverviewPage.jsx)

These are the pages that where the above alerts display:

- https://staging.va.gov/manage-va-debt/summary
- https://staging.va.gov/manage-va-debt/summary/copay-balances
- https://staging.va.gov/manage-va-debt/summary/debt-balances

@Dr-Pongo, can you double check this for me?

**Comment by josephrlee** [comment]
Marking this issue as closed, as this is the design ticket, I will create a follow-up ticket to bring into spring 130 for the frontend work. 

cc: @Tom-Davis @HeatherRienks @maxx1128
